indexer-app/src/cmr/indexer/data/concepts/service.clj

```diff
@@ -38,8 +38,6 @@
        :deleted deleted
        :service-name service-name
        :service-name.lowercase (string/lower-case service-name)
-       :long-name long-name 
-       :long-name.lowercase (string/lower-case long-name)
```

Comment:
I wasn't sure if I should add the long-name here...   I still don't understand what there fields are there for... Could you explain?

Reply:
Actually, that reminds me -- when @chris-durbin and I were looking at this code, he said that the comment in this section was incorrect. I'm going to remove it ...

As to your question, I have some objections to this function and I want to change it, but I'd need to talk to @ygliuvt first. My problem with it is that it collapses two very different behaviours into one function: creating an elastic doc from a concept, and converting a tombstoned concept to something that won't get indexed on things that aren't there anymore. I'd rather separate those bits out into two different private functions ... but the thing I want to check with @ygliuvt is the how keywords are treated in tombstoned concepts.

So, that's a bit of background on the fact that I view the stuff you're asking about as a bit knarly ;-) To answer your question:

These are all the Elastic fields that will get populated when the concept is saved/updated to ES. In the case of deletion (tombstoning), the parsed-concept passed to this function is actually just the `:extra-fields`, and if you look in metadata-db, you'll see that the only extra-fields defined for services is `service-name` ... as such, `long-name` here will be null, and will throw a null exception. Took a while to sort through this (@chris-durbin is the one who found the cause, during a pairing session).

Reply:
Thank you so much for catching this and explaining it.